xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-api/src/main/java/org/xwiki/panels/internal/AbstractPanelsUIExtensionManager.java

```diff
@@ -109,7 +110,18 @@
                     for (UIExtension extension : allExtensions) {
                         // Resolve and serialize the extension id to make it absolute before we can compare it with
                         // the serialized panel reference.
-                        String extensionId = serializer.serialize(resolver.resolve(extension.getId()));
+
+                        String extensionId;
+
+                        // in this case we can directly retrieve the document reference which gives also the wiki name
+                        // allows to fix XWIKI-14557
+                        if (extension instanceof WikiUIExtension) {
```

Comment:
Would be better to manipulate WikiComponent instead of WikiUIExtension since both PanelWikiUIExtension and WikiUIExtension implement it.

Reply:
Hm indeed. I'll use it.